pkg/api/billingdocument.go

```diff
@@ -0,0 +1,34 @@
+package api
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+// BillingDocuments represents billing documents.
+// pkg/database/cosmosdb requires its definition.
+type BillingDocuments struct {
+	Count            int                `json:"_count,omitempty"`
+	ResourceID       string             `json:"_rid,omitempty"`
+	BillingDocuments []*BillingDocument `json:"Documents,omitempty"`
+}
+
+// BillingDocument represents a billing document.
+// pkg/database/cosmosdb requires its definition.
+type BillingDocument struct {
+	MissingFields
+
+	ID          string                 `json:"id,omitempty"`
+	ResourceID  string                 `json:"_rid,omitempty"`
+	Timestamp   int                    `json:"_ts,omitempty"`
+	Self        string                 `json:"_self,omitempty"`
+	ETag        string                 `json:"_etag,omitempty"`
+	Attachments string                 `json:"_attachments,omitempty"`
+	LSN         int                    `json:"_lsn,omitempty"`
+	Metadata    map[string]interface{} `json:"_metadata,omitempty"`
+
+	Billing *Billing `json:"billing,omitempty"`
+
+	OpenShiftClusterID              string `json:"openShiftClusterId,omitempty"`
+	OpenShiftClusterResourceGroupID string `json:"openShiftClusterResourceGroupId,omitempty"`
+	Location                        string `json:"location,omitempty"`
+	TenantID                        string `json:"tenantID,omitempty"`
```

Comment:
can we catch up on these on a call?  I'd like to understand a bit better the drivers for denormalising here.

Reply:
sure I am happy to get on a call.

Reply:
If this is late call, can it be recorded or notes taken? Just for my own understanding and knowledge.